scripts/npn.py

```diff
@@ -21,54 +27,132 @@ def __init__(self, **kwargs):
+        plantae_files = []
+        animalia_files = []
+        request_src = "http://www.data-retriever.org/"
+        base_url = "http://www.usanpn.org/npn_portal/observations/getObservations.xml?start_date={startYear}-01-01&end_date={endYear_date}&request_src={request_src}"
+        start_year = 2009
+        end_year = int(time.strftime("%Y"))
+        header_values = ["observation_id",
+                         "update_datetime",
+                         "site_id",
+                         "latitude",
+                         "longitude",
+                         "elevation_in_meters",
+                         "state",
+                         "species_id",
+                         "genus",
+                         "species",
+                         "common_name",
+                         "kingdom",
+                         "individual_id",
+                         "phenophase_id",
+                         "phenophase_description",
+                         "observation_date",
+                         "day_of_year",
+                         "phenophase_status",
+                         "intensity_category_id",
+                         "intensity_value",
+                         "abundance_value"
+                         ]
+
+        columns = [("record_id", ("pk-auto",)),
+                   ("observation_id", ("int",)),  # subsequently refered to as "status record"
+                   ("update_datetime", ("char",)),
+                   ("site_id", ("int",)),
+                   ("latitude", ("double",)),
+                   ("longitude", ("double",)),
+                   ("elevation_in_meters", ("char",)),
+                   ("state", ("char",)),
+                   ("species_id", ("int",)),
+                   ("genus", ("char",)),
+                   ("species", ("char",)),
+                   ("common_name", ("char",)),
+                   ("kingdom", ("skip",)),  # skip kingdom
+                   ("individual_id", ("char",)),
+                   ("phenophase_id", ("int",)),
+                   ("phenophase_description", ("char",)),
+                   ("observation_date", ("char",)),
+                   ("day_of_year", ("char",)),
+                   ("phenophase_status", ("char",)),
+                   ("intensity_category_id", ("char",)),
+                   ("intensity_value", ("char",)),
+                   ("abundance_value", ("char",))
+                   ]
+        while start_year <= end_year:
+            if start_year == end_year:
+                # get the data in the current year to current day (sometimes the recent day's data is empty)
+                data_url = base_url.format(startYear=start_year, endYear_date=str(time.strftime("%Y-%d-%m")),
+                                           request_src=request_src)
+            else:
+                data_url = base_url.format(startYear=start_year, endYear_date=str(start_year + 1) + "-01-01",
```

Comment:
`endYear_date=str(start_year) + "-12-31"` may avoid the potential for data on January 1st to get duplicated.